### PR TITLE
COMPLETE HTML and CSS practice

### PR DESCRIPTION
**Pull request by karelskiy**
### HTML and CSS practice

[Demo](https://karelskiy.github.io/Close-task-HTML-CSS-practice/) |
[Code base](https://github.com/karelskiy/Close-task-HTML-CSS-practice)

The code is submitted in its own sub-directory and in a dedicated feature branch.
Please, review.

### CODE REVIEW

File: submissions/karelskiy/HTML and CSS practice/index.html [hunk 1]
```diff
@@ -0,0 +1,93 @@
+<!DOCTYPE html>
```
Comment (A-Ostrovnyy) [inline submissions/karelskiy/HTML and CSS practice/index.html]: Check your markup in [W3CValidator](https://validator.w3.org/).
Fix all errors if you have.

File: submissions/karelskiy/HTML and CSS practice/index.html [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+<html lang="en">
+
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <meta http-equiv="X-UA-Compatible" content="ie=edge">
+    <link rel="stylesheet" href="style.css">
+    <link href="https://fonts.googleapis.com/css?family=Raleway&display=swap" rel="stylesheet">
+    <script defer src="script.js"></script>
+    <title>Document</title>
```
Comment (A-Ostrovnyy) [inline submissions/karelskiy/HTML and CSS practice/index.html]: Add the corresponding title of the page.

File: submissions/karelskiy/HTML and CSS practice/index.html [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+</head>
+
+<body>
+    <a href="#"><img id="Doe" width="35px" height='35px'
```
Comment (A-Ostrovnyy) [inline submissions/karelskiy/HTML and CSS practice/index.html]: Good practice for navigation is using `nav` tag and list of menu items.
Also, add some menu items like Google-style.
Avoid using inline styles.

File: submissions/karelskiy/HTML and CSS practice/index.html [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+            title="John Doe" src="img/JhonDoe.jpg"
+            alt=""></a>
```
Comment (A-Ostrovnyy) [inline submissions/karelskiy/HTML and CSS practice/index.html]: Empty `alt` attribute of `img`  tag is bad practice. Fill all `alt` attributes of `img` tags.

File: submissions/karelskiy/HTML and CSS practice/index.html [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+                    <img src="img/pinterest.png">
+                    <figcaption>Pinterest</figcaption>
+                </figure></a>
+            </div>
+            
+        </div>
+    </div>
+</body>
+
+</html>
```
Comment (A-Ostrovnyy) [inline submissions/karelskiy/HTML and CSS practice/index.html]: Add an empty line at the end of each file.

File: submissions/karelskiy/HTML and CSS practice/script.js [hunk 6]
```diff
@@ -0,0 +1,35 @@
+sec.addEventListener('click', function(e){
```
Comment (A-Ostrovnyy) [inline submissions/karelskiy/HTML and CSS practice/script.js]: This task was supposed to be done without JS.
Don't use JS in this task.

File: submissions/karelskiy/HTML and CSS practice/style.css [hunk 7]
```diff
@@ -0,0 +1,153 @@
+ body{
+    margin: 0;
+    font-family: 'Raleway', sans-serif;
+}
+
+.section{
```
Comment (A-Ostrovnyy) [inline submissions/karelskiy/HTML and CSS practice/style.css]: Avoid using `position: absolute;` for defining the grid.

File: submissions/karelskiy/HTML and CSS practice/style.css [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+    top: -19px; 
+    transform: rotate(180deg);
+    z-index: 9000;
+   }
+
+.open{
+    display: block;
+}
+
+#Doe{
```
Comment (A-Ostrovnyy) [inline submissions/karelskiy/HTML and CSS practice/style.css]: Avoid styling by `#id` and HTML tag.
Good practice to separate responsibilities: HTML tags for markup, Class - styling, #ID - JS.

File: submissions/karelskiy/HTML and CSS practice/style.css [hunk 9]
```diff
@@ -0,0 +1,153 @@
...
+    margin-top: -12px; 
+    text-align: center;
+    font-size: .7em;
+}
+
+.container{
+    width: inherit;
+    height: inherit;
+    overflow: auto;
+
```
Comment (A-Ostrovnyy) [inline submissions/karelskiy/HTML and CSS practice/style.css]: Remove unused lines of code.

File: submissions/karelskiy/HTML and CSS practice/index.html [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+                </figure></a>
+                <a href="#"><figure>
+                    <img src="img/crunchyroll.png">
+                    <figcaption>Crunchyroll</figcaption>
+                </figure></a>
+                <a href="#"> <figure>
+                    <img src="img/linkedin.png">
+                    <figcaption>LinkedIn</figcaption>
+                </figure></a>
+                <a href="#" id="more"><span>More</span></a>
```
Comment (A-Ostrovnyy) [inline submissions/karelskiy/HTML and CSS practice/index.html]: Add styles on `:hover` state.

File: submissions/karelskiy/HTML and CSS practice/script.js (hunk 6 above)
Comment (karelskiy) [reply submissions/karelskiy/HTML and CSS practice/script.js]: Thank for comments, at first sorry about js, i didn't saw this. Also i want ask you, about this button, it should be hidden when i forcibly scrolling mouse? Or just click on button without scrolling?

File: submissions/karelskiy/HTML and CSS practice/style.css (hunk 8 above)
Comment (karelskiy) [reply submissions/karelskiy/HTML and CSS practice/style.css]: Thank for this tip

File: submissions/karelskiy/HTML and CSS practice/script.js (hunk 6 above)
Comment (A-Ostrovnyy) [reply submissions/karelskiy/HTML and CSS practice/script.js]: As I understand you right, this button must be present always, also, it must be accessible from the keyboard.

File: submissions/karelskiy/HTML and CSS practice/index.html [hunk 11]
```diff
@@ -7,87 +7,117 @@
...
-                <a href="#" class="hidden"> <figure>
-                    <img src="img/xing.png">
-                    <figcaption>Xing</figcaption>
-                </figure></a>
-                <a href="#" class="hidden"><figure>
-                    <img src="img/pinterest.png">
-                    <figcaption>Pinterest</figcaption>
-                </figure></a>
+    <nav class="nav">
+        <a href="#" class="lImg li" tabindex="1">Images</a>
```
Comment (A-Ostrovnyy) [inline submissions/karelskiy/HTML and CSS practice/index.html]: Good practice to use `nav>ul>li>a` structure to defining navigation.

File: submissions/karelskiy/HTML and CSS practice/index.html [hunk 12]
```diff
@@ -7,87 +7,117 @@
...
+        <a href="#" class="lMail li" tabindex="2">Mail</a>
+        <a href="#" class="lDoe li"><img class="img-Doe" src="img/JhonDoe.jpg"
```
Comment (A-Ostrovnyy) [inline submissions/karelskiy/HTML and CSS practice/index.html]: All interactive elements have focus state in the browser by default. Using `tabindex` you redefine default browser order of focused elements. Be careful redefining the default focus order.

File: submissions/karelskiy/HTML and CSS practice/style.css [hunk 13]
```diff
@@ -1,94 +1,106 @@
...
-    height: 2px;
-    background-color: gray;
-}
-
-.menu-btn span::before,
-.menu-btn span::after{
-    content: '';
-    
+
+a.lImg, a.lMail, .fig a{
```
Comment (A-Ostrovnyy) [inline submissions/karelskiy/HTML and CSS practice/style.css]: Don't use too much-specific selectors. Try to use as simplest selector as it needs.

File: submissions/karelskiy/HTML and CSS practice/index.html (hunk 12 above)
Comment (karelskiy) [reply submissions/karelskiy/HTML and CSS practice/index.html]: As i understand correctly, i need remove this tabindex?

Comment (A-Ostrovnyy) [reply submissions/karelskiy/HTML and CSS practice/index.html]: I this case you don't need `tabindex`
[Read more](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/tabindex) about `tabindex` why and how to use it.

Comment (karelskiy) [reply submissions/karelskiy/HTML and CSS practice/index.html]: Or, i should add in all elements?

Comment (karelskiy) [reply submissions/karelskiy/HTML and CSS practice/index.html]: OK, thanks.

**Comment by A-Ostrovnyy** [comment]
Well done

**Pull request merged**